src/run_program.rs

```diff
@@ -475,8 +475,7 @@ impl<'a, D: Dialect> RunProgramContext<'a, D> {
             if cost > effective_max_cost {
                 return err(max_cost_ptr, "cost exceeded");
             }
-            let top = self.op_stack.pop();
-            let op = match top {
+            let op = match self.op_stack.pop() {
```

Comment:
I suspect this whole `loop` should probably be turned into something like:
```
while let Some(op) = self.op_stack.pop() {
   ...
}
```

as long as we can ensure the cost check is still performed in the edge case of it exceeding on the last operator